source/iopipe/json/serialize.d

```diff
@@ -1132,8 +1162,220 @@ private void deserializeImplWithPolicy(T, JT, Policy)(
         
 }
 
+private void deserializeImplWithPolicy(T, JT, Policy)(
+    ref Policy policy,
+    ref JT tokenizer,
+    ref T item
+) if (__traits(isStaticArray, T))
+{
+    auto context = onArrayBegin(policy, tokenizer, item);
+    auto jsonItem = tokenizer.nextSignificant
+        .jsonExpect(JSONToken.ArrayStart, "Parsing " ~ T.stringof);
+
+    // Parse array elements
+    size_t elementCount = 0;
+    while(true) {
+         
+        policy.onArrayElement(tokenizer, item, elementCount, context);
+        elementCount++;
+
+        if (tokenizer.peekSignificant() == JSONToken.ArrayEnd) {
+            // If we hit the end of the array, break
+            break;
+        }
+
+        jsonItem = tokenizer.nextSignificant(); // consume the comma
```

Comment:
This needs to verify that it's a comma. For instance, if you get an unexpected token here, this will cause issues. Yes, the tokenizer is giving you only a valid sequence, but it also can hand you errors.